### DISCUSSION TITLE
New Intelliflo3

### DISCUSSION THREAD

**Discussion by katodude** [tagyoureit/nodejs-poolController#864] [RS485] [answered]
Just replaced my Intelliflo with a Intelliflo3. Also upgraded the equipment to a Pi4, MegaBas, and a new 8 relay card. 

Cannot get the RS-485 to work. The two jumpers are on the MegaBas. Serial port is enabled but not the serial console (made that mistake once). Reversed the wires a whole bunch of times. 

Attached is a log. 
[replay 2.zip](https://github.com/tagyoureit/nodejs-poolController/files/13296692/replay.2.zip)

**Comment by rstrouse** [comment]
Ok the port is silent for both the Chlorinator and the pump.  It also looks like it found the port and opened it.  Please verify this in dashPanel on this page.  It should show as open for the status.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/1f72de2e-13df-4857-a89d-9fd4fc542903)

Next download the megabas-rpi repo and update the firmware.  But whatever you do shut down REM before performing the update.  It will fail in odd ways if the card is accessed during the update.   Also, make sure you do not have the RS485 jumpers installed on the Relay 8,
https://github.com/SequentMicrosystems/megabas-rpi

After you have updated the firmware run the following commands
`cd ~/megabas-rpi`
`megabas 0 rs485wr 0 9600 1 0 1`



`

**Comment by katodude** [comment]
It does show as open. Also the chlorinator is not attached so thats why no comms there. I should probably remove that. 

Will do the updates tonight.

**Comment by katodude** [comment]
Did the update. No change. No jumpers on the 8relay. 

I may just try a new cable tomorrow.

**Comment by katodude** [comment]
Well it’s not the cable. Tried a new cable, and then tested the continuity on the old cable.

**Comment by rstrouse** [comment]
https://github.com/tagyoureit/nodejs-poolController/discussions/844

There were a couple of other users that were having issues with the BAS RS485 port.  One ended up using the RS485 port on the Relay 8.  Probably not a bad idea.

Also, just to be sure.  Verify that you do not have the serial console enabled and disable Bluetooth.

**Comment by katodude** [comment]
Double checked Serial Console is disabled. Bluetooth was enabled but just disabled it. 

I guess I will try the relay board this weekend. I guess I just remove the jumpers from the Bas and put them on the relay card and connect to the 485 port there. 

Did one more dump just in case. 
[replay 3.zip](https://github.com/tagyoureit/nodejs-poolController/files/13312727/replay.3.zip)

**Reply by rstrouse** [reply]
Oh I forgot something.  First, you must not be connected to it via wifi.  Also after disconnecting from the pump you must press the stop button to shut it off.  Then turn it back on.  It will go through and turn the connection magenta after that and disable the external connections.

**Comment by katodude** [comment]
Well, I factory reset the pump so it is not WiFi connected. 

I removed the 8relay board to make sure it was not interfering. 

Switched the wires around twice, and used the stop button to cycle the pump on and off. Still no magenta and no control. The com button on the pump is flashing orange. 

I am ready to try the RS485 on the 8relay board. Other than install the jumpers on the tx/rx are there any other settings?

**Comment by rstrouse** [comment]
Nope no other settings

**Comment by katodude** [comment]
Well, I am really confused now. It’s not the pump. 

I took an old Rpi 3 that I was using before. Hooked up a USB RS-485 and attached that to the pump. Went magenta in a few seconds. Took that same USB with the cable still attached and plugged it into the Rpi4 with the boards on it and still nothing. 

The USB has two led that flash (I guess when there is a TX/RX) and they dont even flash when attached to the RPI4.

Is there some setting in a RPI4 that is different that I am missing?

**Reply by rstrouse** [reply]
Interesting.  Even the usb port is not working?  You aren’t running the 64 bit os are you?

**Comment by katodude** [comment]
Just double checked. I ran a getconf LONG_BIT

It came back as 32.

**Reply by rstrouse** [reply]
Something is definitely not right with the PI, OS, or config.  Especially if you cannot get a working USB setup to talk on the TX line.  Are you sure you do not have remote GPIO or the Serial Console enabled?

**Comment by katodude** [comment]
Double and tripled checked those settings. Anyway here is the plan of action. Ordered a new SD card that I will image today. Also ordered a new Pi which was supposed to be delivered yesterday, but might come today. 

Will test that out with a USB port first. Then rebuild the controller piece by piece with the new Pi, testing at every step.

**Comment by katodude** [comment]
Well the Pi arrived a little early. Flashed a new build. Setting shown below. 

I did insert the USB RS-485 that has the LEDs on it. The LEDs flash when starting up but then stop. On the Pi3 there is always 1 that is flashing slowly even when not connected to anything. I have not tried hooking it up to the pump yet. But it the lack of flashing concerns me that something is wrong with OS or the way it works on a Pi4. 

The Config in the UI has sliders instead of pushbuttons? Does that mean anything?

![IMG_0258](https://github.com/tagyoureit/nodejs-poolController/assets/66255504/092fc82c-aec3-4b10-99f3-1cf0e2e4314f)

**Comment by katodude** [comment]
Also checking again on the Pi3. The pool controller has send stats. All failed when not connected but trying to send. On the Pi4 there are not attempted sends.

**Comment by katodude** [comment]
Wait…..

Ignore the last stuff. Forgot to add a pump to the Nixie. USB is now flashing. 

Going to the pump to test it.

**Comment by katodude** [comment]
Ok, Stand alone RPi4 with USB is now communicating with the pump. 

Time to tear down the box and replace the new old Pi with the new.

**Comment by truvec** [comment]
I have two pumps IF3 VSF 011075 and IF VSF 011056. I am using mega-bas and The Intelliflo 3 is giving me many hours of trouble. The pump will go magenta (automation mode) and will set speed. But Nixie shows all sent as dropped or abandoned. No successful sends. It gets telemetry from the pump. 

So I decided to hook up the 011056. Sends and receives - No problems.  

It seems that the Pump is the variable, not the mega-bas. Waiting on a usb 485 to see if something goes right.

Have you made progress with this?

**Comment by katodude** [comment, marked as answer]
I have had success. 

It turns out it was the cable. The cable worked perfectly fine with the old pump and not with the new one. Replaced the cable and now all is good. It’s weird, but thats what happened.

**Reply by truvec** [reply]
Seems like it’s always the effing cable.  I’m going to check that. Thanks
